docarray/array/doc_list/io.py

```diff
@@ -515,6 +514,8 @@ class Person(BaseDoc):
             doc_dict = _access_path_dict_to_nested_dict(access_path2val)
             docs.append(doc_type.parse_obj(doc_dict))
 
+        if not isinstance(docs, cls):
+            return cls(docs)
```

Comment:
How could it not be from instance `cls` ?

Reply:
`docs` can also be a list of Documents etc.